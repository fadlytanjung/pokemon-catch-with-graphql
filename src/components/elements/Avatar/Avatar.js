import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Avatar(props) {
  const { className, src } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');

  return <figure className={customClass} style={{ backgroundImage: `url('${src}')` }} />;
}

Avatar.defaultProps = {
  className: '',
  src: '',
};

Avatar.propTypes = {
  className: PropTypes.string,
  src: PropTypes.string,
};
