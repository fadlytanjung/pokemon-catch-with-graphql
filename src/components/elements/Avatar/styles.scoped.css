.root {
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  border-radius: 50%;
  height: 2rem;
  width: 2rem;
}