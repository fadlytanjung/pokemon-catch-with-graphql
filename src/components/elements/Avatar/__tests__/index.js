import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Avatar from '../Avatar';

describe('src/components/elements/Avatar', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Avatar />);
    expect(tree).toMatchSnapshot();
  });
});
