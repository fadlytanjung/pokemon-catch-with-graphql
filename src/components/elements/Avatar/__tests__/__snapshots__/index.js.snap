// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Avatar render 1`] = `
<figure
  className=""
  style={
    Object {
      "backgroundImage": "url('')",
    }
  }
/>
`;
